File: aspnet-core/MB.STA/MB.STA.Entities/Booking.cs
using System;
using System.Collections.Generic;

namespace MB.STA.Entities
{
    public class Booking
    {
        public Booking()
        {
            Passengers = new List<Passenger>();
        }

        public int Id { get; set; }
        public DateTime PickUpTime { get; set; }
        public string FromAddress { get; set; }
        public string ToAddress { get; set; }
        public double Price { get; set; }
        public bool IsPaid { get; set; }
        public DateTime? PaymentDate { get; set; }

        public int CarId { get; set; }
        public Car Car { get; set; }

        public int DriverId { get; set; }
        public Driver Driver { get; set; }

        public List<Passenger> Passengers { get; set; }
    }
}

File: angular/src/app/shared/models/Booking.ts
import { Car } from "./Car";
import { Driver } from "./Driver";
import { Passenger } from "./Passenger";

export interface Booking {
    id: number;
    pickUpTime: Date;
    fromAddress: string;
    toAddress: string;
    price: number;
    isPaid: boolean;
    paymentDate: Date;
    carId?: number;
    car: Car;
    driverId?: number;
    driver: Driver;
    passengers: Passenger[]
}